test/e2e/bgppolicy_test.go

```diff
@@ -130,77 +123,193 @@ func TestBGPPolicy(t *testing.T) {
+
+		t.Log("Verify the connectivity of the installed routes on remote FRR route")
+		ipsToConnect := []string{podIP, clusterIP}
+		for _, ip := range ipsToConnect {
+			cmd := fmt.Sprintf("/usr/bin/wget -O - http://%s:8080/hostname -T 5", ip)
+			rc, stdout, _, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, cmd, "/", nil, "")
+			require.NoError(t, err)
+			require.Equal(t, 0, rc)
+			require.Equal(t, podName, stdout)
+		}
+
+		t.Log("Update the BGP configuration on the remote FRR router")
+		configureExternalBGPRouter(t, externalASN, updatedNodeASN, false)
+
+		t.Log("Update InternalTrafficPolicy of the test Service Cluster to Local")
+		_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
+		defer func() {
+			data.updateServiceInternalTrafficPolicy("agnhost-svc", false)
+		}()
 		require.NoError(t, err)
-		require.Equal(t, 0, rc)
-		require.Equal(t, podName, stdout)
-	}
 
-	t.Log("Update the BGP configuration on the remote FRR router")
-	configureExternalBGPRouter(t, externalASN, updatedNodeASN, false)
+		t.Log("Update the test BGPPolicy with a new local ASN and removing Pod CIDR advertisement")
+		updatedBGPPolicy := bgpPolicy.DeepCopy()
+		updatedBGPPolicy.Spec.LocalASN = updatedNodeASN
+		updatedBGPPolicy.Spec.Advertisements.Pod = nil
+		_, err = data.crdClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
+		require.NoError(t, err)
 
-	_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
-	require.NoError(t, err)
+		t.Log("Get routes installed on remote FRR router and verify them")
+		expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
+		notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
+		checkFRRRouterBGPRoutes(t, expectedRoutes, notExpectedRoutes)
+
+		t.Log("verify the connectivity of the installed routes on remote FRR route")
+		ipsToConnect = []string{clusterIP}
+		for _, ip := range ipsToConnect {
+			cmd := fmt.Sprintf("/usr/bin/wget -O - http://%s:8080/hostname -T 5", ip)
+			rc, stdout, _, err := exec.RunDockerExecCommand(externalInfo.externalFRRCID, cmd, "/", nil, "")
+			require.NoError(t, err)
+			require.Equal(t, 0, rc)
+			require.Equal(t, podName, stdout)
+		}
+	})
 
-	t.Log("Update the test BGPPolicy with a new local ASN")
-	updatedBGPPolicy := bgpPolicy.DeepCopy()
-	updatedBGPPolicy.Spec.LocalASN = updatedNodeASN
-	updatedBGPPolicy.Spec.Advertisements.Pod = nil
-	_, err = data.crdClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
-	require.NoError(t, err)
+	t.Run("Multiple BGPPolies applied to different Nodes within a confederation", func(t *testing.T) {
+		t.Log("Create a test BGPPolicy selecting all Nodes as well as advertising ClusterIPs and Pod CIDRs")
+		const confederationIdentifier = int32(60000)
+		const updatedConfederationIdentifier = int32(60001)
+		asnStart := int32(61000)
```

Comment:
based on usage, I am not sure why you use a variable here while you use constants for confederation identifiers.

Reply:
Removed all the constants since it looks weird to use variable and constant together here.